scripts/release_scripts/initial_release_prep_test.py

```diff
@@ -135,56 +133,59 @@ def mock_run_cmd(cmd_tokens):
     def test_cut_release_branch_with_correct_version(self):
         check_function_calls = {
             'open_new_tab_in_browser_if_possible_is_called': False,
-            'check_call_is_called': False
+            'execute_branch_cut_is_called': False
         }
         expected_check_function_calls = {
             'open_new_tab_in_browser_if_possible_is_called': True,
-            'check_call_is_called': True
+            'execute_branch_cut_is_called': True
         }
         def mock_open_tab(unused_url):
             check_function_calls[
                 'open_new_tab_in_browser_if_possible_is_called'] = True
-        def mock_check_call(unused_cmd_tokens):
-            check_function_calls['check_call_is_called'] = True
+        def mock_require_cwd_to_be_oppia():
+            check_function_calls['execute_branch_cut_is_called'] = True
+            raise Exception('Exception raised to stop branch cut.')
         def mock_input():
             return '1.2.3'
 
         open_tab_swap = self.swap(
             common, 'open_new_tab_in_browser_if_possible',
             mock_open_tab)
-        check_call_swap = self.swap(
-            subprocess, 'check_call', mock_check_call)
+        cwd_swap = self.swap(
+            common, 'require_cwd_to_be_oppia', mock_require_cwd_to_be_oppia)
         input_swap = self.swap(
             python_utils, 'INPUT', mock_input)
-        with open_tab_swap, check_call_swap, input_swap:
+        with open_tab_swap, cwd_swap, input_swap, self.assertRaisesRegexp(
+            Exception, 'Exception raised to stop branch cut.'):
```

Comment:
What's catching these exceptions?

Also, why do we need to raise them? Seems like an odd pattern.

Reply:
Some context on why I made this change ([Ref](https://github.com/oppia/oppia/pull/8577#discussion_r379821699)): 

- I updated the arg passed in `cut_release_or_hotfix_branch` from `new_version` to `release_version` in PR #8577 
- I missed changing the subprocess call in `initial_release_prep` and the tests did not catch it since I was mocking `subprocess.check_call` in tests.
- So, I need to actually run the `cut_release_or_hotfix_branch` script instead of mocking `subprocess.check_call` in the tests but also ensure that branch cut is not executed. 
- If I mocked the `execute_branch_cut` function which is being called, it will again fail to detect if a wrong arg is passed.
- So, I wanted the function to run but stop as soon as arg check is done. The first line after arg check in the function is to check if the cwd is oppia. So, I mocked that function instead and raised an Exception there to ensure that args are checked as well as the function stops execution after that.

\cc @DubeySandeep

Reply:
You can ignore the previous comment. I have modified the branch cut script by adding a main method as per @DubeySandeep's suggestion and now there is no need for exception.